### PR TITLE
Feat/wallet balances

### PR DESCRIPTION
**Pull request by mjuchli-da**

### CODE REVIEW

File: gui-backend/src/main/java/com/rln/gui/backend/implementation/methods/BalancesApiImpl.java [hunk 1]
```diff
@@ -69,26 +71,40 @@ public List<Balance> getAddressBalance(String address) throws IbanNotFoundExcept
...
                 .orElseThrow(() -> new IbanNotFoundException(address));
 
-        var builder = Balance.builder()
-            .address(address)
-            .assetId(0L) // default to 0 now, as we don't have assetId in the system yet
-            .assetName(assetName);
+        var builder = accountCache
+                .getAccountInfo(address)
+                .map(info -> {
+                    Balance.AssetOrLiabilityEnum assetOrLiabilityEnum;
```
Comment (gaborh-da) [inline gui-backend/src/main/java/com/rln/gui/backend/implementation/methods/BalancesApiImpl.java:77]: Maybe this small piece about the liability vs asset thing could be extracted into a function.

File: gui-backend/src/main/java/com/rln/gui/backend/implementation/methods/BalancesApiImpl.java [hunk 2]
```diff
@@ -120,20 +136,40 @@ public void delete(String provider, String address) throws NonZeroBalanceExcepti
...
+                .map(AccountInfo::getIban)
+                .map(this::getAddressBalance)
+                .flatMap(List::stream)
+                .collect(Collectors.toList());
     }
 
-    public Object testWalletAddress(String address,
-        @Valid @NotNull WalletAddressTestDTO walletAddressTestDTO) {
-        return null;
+    private Balance toBalance(com.rln.damlCodegen.model.balance.Balance contract) {
```
Comment (gaborh-da) [inline gui-backend/src/main/java/com/rln/gui/backend/implementation/methods/BalancesApiImpl.java:162]: Where do we use this?

File: gui-backend/src/main/java/com/rln/gui/backend/implementation/GuiBackendApiImplementation.java [hunk 3]
```diff
@@ -162,7 +147,10 @@ public void postWallets(@Valid @NotNull WalletDTO walletDTO) {
   // Get all the balances for all addresses in a wallet
   @Override
   public List<Balance> getBalances(Long walletId) {
-    throw notImplemented();
+    if (!walletId.equals(ONLY_SUPPORTED_WALLET.getId()))
```
Comment (gaborh-da) [inline gui-backend/src/main/java/com/rln/gui/backend/implementation/GuiBackendApiImplementation.java:150]: One small ask:
```
if (!walletId.equals(ONLY_SUPPORTED_WALLET.getId())) {
  throw notFound();
}
```
Please add the parentheses if not a problem :)

File: gui-backend/src/main/java/com/rln/gui/backend/implementation/methods/BalancesApiImpl.java (hunk 2 above)
Comment (mjuchli-da) [reply gui-backend/src/main/java/com/rln/gui/backend/implementation/methods/BalancesApiImpl.java:162]: It's an old snippet. Removed.

File: gui-backend/src/main/java/com/rln/gui/backend/implementation/methods/BalancesApiImpl.java (hunk 1 above)
Comment (mjuchli-da) [reply gui-backend/src/main/java/com/rln/gui/backend/implementation/methods/BalancesApiImpl.java:77]: Done

**Review by gaborh-da — APPROVED** [review]

**Pull request merged**